### DISCUSSION TITLE
NextJS new install not loading on localhost:3000 (partially solved)

### DISCUSSION THREAD

**Discussion by kirilchristov** [vercel/next.js#52269] [Help]
### Summary

A fresh new app is not loading on localhost:3000. Endless loading spinner. (Kinda solved, read the update at the bottom)

I have a recent situation where none of my next projects are working. I did all the steps from the official [setup](https://nextjs.org/learn/basics/create-nextjs-app/setup) with no luck.

I get this message in the terminal:
> - ready started server on 0.0.0.0:3000, url: http://localhost:3000
>- event compiled client and server successfully in 146 ms (18 modules)

**What I have tried so far:**

1. Delete .next dir
2. Restarted my Mac
3. Tried 4 different node versions: 18.12.1, 18.16.0,19.9.0 and the latest 20.4.0 (I cannot update more than that) - I use nvm for versioning
4. Deleted and reinstalled all node modules
5. Changed ports from 3000 -> 3005, etc...

Any reason that might comes to mind?

### Additional information

**UPDATE**:
I have just discovered that the issue is using a VPN configuration on the Mac. That causes issues with connecting to localhost. I am trying to figure out how to solve it. For now you can just disable the vpn and restart your server. After you access port 3000 you can enable the VPN again, it will work (surprisingly, at least for me). 

_No response_

**Comment by icyJoseph** [comment]
Sounds like an architecture problem. I've seen this before. 

Try running: `node -p 'process.arch'` what does it output?

Let's start with that.

**Reply by kirilchristov** [reply]
arm64... but I think I just got it.... it is the VPN :(

**Reply by icyJoseph** [reply]
Ah... darn strange though, so that means you can't reach your own 3000? port? What happens if you go to 0.0.0.0:3000 or 127.0.0.1:3000?

**Reply by kirilchristov** [reply]
Not working. Tried both. But once you start the app you can turn on the VPN
 again. What I do is I just turn off the wifi. But it is not a solution.

**Reply by icyJoseph** [reply]
What VPN is this? Maybe it has some kind of protection rule to forbid ports? Next.js uses workers through port sockets to process pages, and such, typically it uses a randomly assigned port in the 5xxxx range, maybe that's not being let through by the VPN?

**Reply by kirilchristov** [reply]
NordVPN with dedicated IP
